### PR TITLE
Update/component base

### PR DESCRIPTION
**Pull request by davidesner**
Added ComponentBase to be used in Python Template.

### CODE REVIEW

**Comment by davidesner** [comment]
@bakobako check this out, does it make sense to you?

**Comment by bakobako** [comment]
@davidesner could the default value of required_image_parameters be [] . As it is usually empty anyways, the None forces me to always set it to an empty list

File: src/keboola/component/base.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                optional path to data folder that overrides the default behaviour (`KBC_DATADIR` environment variable).
+                May be also specified by '-d' or '--data' commandline argument
+        Raises:
+            UserException - on config validation errors.
+        """
+
+        # for easier local project setup
+        super().__init__(data_folder_path=self._get_data_folder_override_path(data_path_override))
+
+        self._required_parameters = required_parameters if required_parameters else []
```
Comment (davidesner) [inline src/keboola/component/base.py:42]: @bakobako the parameters are now truly optional

**Review by bakobako — APPROVED** [review]
Looks good

**Pull request merged**